03-rnaseq/differential-expression_rnaseq_01.Rmd

```diff
@@ -0,0 +1,417 @@
+
+In this analysis, we will be using DESeq2 [@DESeq2].
+
+```{r}
+if (!("DESeq2" %in% installed.packages())) {
+  # Install this package if it isn't installed yet
+  BiocManager::install("DESeq2", update = FALSE)
+}
+```
+
+Attach the libraries we need for this analysis:
+
+```{r}
+# Attach the DESeq2 library
+library(DESeq2)
+
+# Attach the ggplot2 library for plotting
+library(ggplot2)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import data and metadata
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_SRP078441.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "SRP078441.tsv" # Replace with the name of your data file
+)) %>%
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+The information we need to make the comparison is in the `refinebio_title` column of the metadata data.frame.
+
+```{r}
+head(metadata$refinebio_title)
+```
+
+## Set up metadata
+
+We'll use some tidyverse functionality to extract this information into its own column to make it easier to use.
+
+```{r}
+metadata <- metadata %>%
+  # the last bit of the title, separated by "-" contains the mutation
+  # information that we want to extract
+  dplyr::mutate(asxl_mutation_status = stringr::word(refinebio_title,
+    -1,
+    sep = "-"
+  )) %>%
+  # now let's make this simpler
+  dplyr::mutate(asxl_mutation_status = dplyr::case_when(
+    grepl("ASXL1|ASXL2", asxl_mutation_status) ~ "asxl_mutation",
+    grepl("ASXLwt", asxl_mutation_status) ~ "no_mutation"
+  ))
+```
+
+Let's take a look at `metadata_df` to see if this worked.
+
+```{r}
+# looking at the first 6 rows of the metadata_df and only at the columns that
+# contain the title and the mutation status we extracted from the title
+head(dplyr::select(metadata, refinebio_title, asxl_mutation_status))
+```
+
+Before we set up our model in the next step, we want to check if our modeling variable is set correctly. 
+We always want our "control" to to be set as the first level in the variable we provide as our experimental variable. 
+
+```{r}
+str(metadata$asxl_mutation_status)
```

Comment:
Why `str()` ? It seems unnecessary to illustrate your point and there is no explanation around it, so I am inclined to take it out.

Reply:
It is necessary to show them that this is a character and not a factor (what my next sentence after this said). But I didn't explain before doing the step, so I see that that is unclear.

Reply:
I added words to explain this step some more.